homeassistant/components/frontier_silicon/media_player.py

```diff
@@ -270,3 +273,42 @@ async def async_select_source(self, source):
     async def async_select_sound_mode(self, sound_mode):
         """Select EQ Preset."""
         await self.fs_device.set_eq_preset(self.__sound_modes_by_label[sound_mode])
+
+    async def async_browse_media(self, media_content_type=None, media_content_id=None):
+        """Browse media library and preset stations."""
+        if media_content_type in (None, "library"):
+            return await browse_top_level(self._attr_source, self.fs_device)
+
+        return await browse_node(self.fs_device, media_content_type, media_content_id)
+
+    async def async_play_media(self, media_type, media_id, **kwargs):
+        """Play selected media or channel."""
+        supported_media_types = [MEDIA_TYPE_CHANNEL, MEDIA_TYPE_PRESET]
+
+        if media_type not in supported_media_types:
+            _LOGGER.error(
+                "Got %s, but frontier_silicon only supports playing media types: %s",
+                media_type,
+                supported_media_types,
+            )
+            return
+
+        keys = media_id.split("/")
+
+        desired_source = keys[0]
+        await self.async_select_source(desired_source)  # this also powers on the device
+
+        if media_type == MEDIA_TYPE_PRESET:
+            if len(keys) != 2:
+                raise BrowseError("Presets can only have 1 level")
+
+            # Keys of presets are 0-based, while the list shown on the device starts from 1
+            preset = int(keys[-1]) - 1
```

Comment:
Writing keys[-1] seems to be a bit convoluted way of writing keys[1] since we already know len(keys) == 2 at this point.